ObjectTracking/src/bin/basic_control_1d.rs

```diff
@@ -172,8 +174,142 @@ impl<T: ToInputArray> ToInputArrayExt for T {
+    fn register_callsite(&self, metadata: &tracing::Metadata<'_>) -> Interest {
+        Interest::sometimes()
+    }
+
+    // If the mjpeg server is behind, disable subscription for a bit
+    fn enabled(&self, metadata: &tracing::Metadata<'_>) -> bool {
+        match self.visitor.try_lock() {
+            Ok(visitor) => !visitor.mjpeg_server.is_full(),
+            _ => false,
+        }
+    }
+
+    fn new_span(&self, span: &span::Attributes<'_>) -> span::Id {
+        todo!("implement new_span({:?})", span)
+    }
+
+    fn record(&self, span: &span::Id, values: &span::Record<'_>) {
+        todo!("implement record({:?}, {:?})", span, values)
+    }
+
+    fn record_follows_from(&self, span: &span::Id, follows: &span::Id) {
+        todo!("implement record_follows_from({:?}, {:?})", span, follows)
+    }
+
+    fn event(&self, event: &tracing::Event<'_>) {
+        match self.visitor.try_lock() {
+            Ok(ref mut visitor) => event.record(visitor.deref_mut()),
+            // visitor is already behind, so let's skip logging for now.
+            _ => (),
+        }
+    }
+
+    fn enter(&self, span: &span::Id) {
+        todo!("implement enter({:?})", span)
+    }
+
+    fn exit(&self, span: &span::Id) {
+        todo!("implement exit({:?})", span)
+    }
+}
+
+struct MJpegVisitor {
+    mjpeg_server: std::sync::Arc<mjpeg_rs::MJpeg>,
+}
+
+impl MJpegVisitor {
+    fn new() -> Self {
+        Self {
+            mjpeg_server: std::sync::Arc::new(mjpeg_rs::MJpeg::new()),
+        }
+    }
+}
+
+impl tracing::field::Visit for MJpegVisitor {
+    fn record_value(&mut self, field: &tracing::field::Field, value: &dyn valuable::Valuable) {
+        value.visit(self)
+    }
+
+    fn record_debug(&mut self, field: &tracing::field::Field, value: &dyn std::fmt::Debug) {
+        todo!()
+    }
+}
+
+impl valuable::Visit for MJpegVisitor {
+    fn visit_named_fields(&mut self, named_values: &valuable::NamedValues<'_>) {
+        todo!("visit_named_fields({:?})", named_values);
+        let _ = named_values;
+    }
+
+    fn visit_unnamed_fields(&mut self, values: &[valuable::Value<'_>]) {
+        for value in values {
+            if let Some(value) = value.as_listable() {
+                value.visit(self)
+            }
+        }
+    }
+
+    fn visit_primitive_slice(&mut self, slice: valuable::Slice<'_>) {
+        if let valuable::Slice::U8(data) = slice {
+            self.mjpeg_server.update_jpeg(data.to_vec()).unwrap()
+        }
+        // todo!("visit_primitive_slice()");
+    }
+
+    fn visit_entry(&mut self, key: valuable::Value<'_>, value: valuable::Value<'_>) {
+        todo!("visit_entry({:?}, {:?})", key, value);
+        let _ = (key, value);
+    }
+
+    fn visit_value(&mut self, value: valuable::Value<'_>) {
+        todo!()
+    }
+}
+
 fn main() -> eyre::Result<()> {
     color_eyre::install()?;
+    // tracing_subscriber::fmt::init();
+    MJpegTracingSubscriber::new().init();
```

Comment:
we need to think about how our computer vision tracing interacts with other types of tracing. For example, what happens if someone enables standard logging and accidentally enables a cv log that contains a huge image every millisecond. I expect we can make the cv log more succinct (currently it uses the debug impl of the Vec<u8> that contains the jpeg), but I doubt we can make it cheap.